src/OptimizeImageTrait.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the ekino/tiny-png-sonata-media-bundle project.
+ *
+ * (c) Ekino
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Ekino\TinyPngSonataMediaBundle;
+
+use Ekino\TinyPngSonataMediaBundle\Consumer\OptimizeImageConsumer;
+use Sonata\NotificationBundle\Backend\BackendInterface;
+use Sonata\NotificationBundle\Backend\QueueDispatcherInterface;
+use Symfony\Component\HttpFoundation\Request;
+
+/**
+ * Trait OptimizeImageTrait.
+ *
+ * @author William JEHANNE <william.jehanne@ekino.com>
+ */
+trait OptimizeImageTrait
+{
+    /**
+     * @var BackendInterface
+     */
+    private $backend;
+
+    public function optimizeAction(Request $request = null)
+    {
+        $this->getBackend()->createAndPublish(OptimizeImageConsumer::CONSUMER_TYPE, [
+            'mediaId' => $request->get('id'),
+        ]);
+
+        $this->get('session')->getFlashBag()->add(
+            'sonata_flash_success',
+            'Successfully launch the optimization of your media.'
+        );
+
+        return $this->redirectToList();
+    }
+
+    /**
+     * @return BackendInterface
+     */
+    private function getBackend()
+    {
+        return $this->container->get('sonata.notification.backend');
```

Comment:
this has to be done by a setter injection. Not from the container anymore, anywhere

Reply:
What's wrong @qsomazzi? 🙂

Reply:
`$this->container->get(` this :)

Reply:
I though you were not agree with the confused emoji

Reply:
I just didn't succeed to put the emoji on the code, but yes i agree with @bmaziere